state/status.go

```diff
@@ -386,14 +400,32 @@ func probablyUpdateStatusHistory(db Database, globalKey string, doc statusDoc) {
 		if current.Status == doc.Status &&
 			current.StatusInfo == doc.StatusInfo &&
 			dataSame(current.StatusData, doc.StatusData) {
-			return
+			historyW := history.Writeable()
+			err = historyW.Update(
+				bson.D{
+					// Since status history records do not have a
```

Comment:
They do have an _id field, but it isn't read into the structure. Perhaps don't serialize into a historicalStatusDoc structure, but one that also includes the _id field?